_posts/2023-09-25-announcing-duckdb-090.md

```diff
@@ -0,0 +1,158 @@
+     alt="Image of the Yellow Billed Duck"
+     width=200px
+     />
+
+The DuckDB team is happy to announce the latest DuckDB release (0.9.0). This release is named Undulata after the [Yellow-billed duck](https://en.wikipedia.org/wiki/Yellow-billed_duck) native to Africa.
+
+To install the new version, please visit the [installation guide](https://duckdb.org/docs/installation/index). The full release notes can be found [here](https://github.com/duckdb/duckdb/releases/tag/v0.9.0).
+
+<!--more-->
+
+#### What's new in 0.9.0
+
+There have been too many changes to discuss them each in detail, but we would like to highlight several particularly exciting features! 
+
+* Out-Of-Core Hash Aggregate
+* Storage Improvements
+* Index Improvements
+* DuckDB-WASM Extensions
+* Extension Auto-Loading
+* Improved AWS Support
+* Iceberg Support
+* Azure Support
+* PySpark-Compatible API
+
+Below is a summary of those new features with examples, starting with a change in our SQL dialect that is designed to produce more intuitive results by default.
+
+
+#### Breaking SQL Changes
+
+[**Struct Auto-Casting**](https://github.com/duckdb/duckdb/pull/8942). Previously the names of struct entries were ignored when determining auto-casting rules. As a result, struct field names could be silently renamed. Starting with this release, this will result in an error instead.
+
+```sql
+CREATE TABLE structs(s STRUCT(i INT));
+INSERT INTO structs VALUES ({'k': 42});
+-- Mismatch Type Error: Type STRUCT(k INTEGER) does not match with STRUCT(i INTEGER). Cannot cast STRUCTs with different names
+```
+
+Unnamed structs constructed using the `ROW` function can still be inserted into struct fields.
+
+```sql
+INSERT INTO structs VALUES (ROW(42));
+```
+
+#### Core System Improvements
+
+**[Out-Of-Core Hash Aggregates.](https://github.com/duckdb/duckdb/pull/7931)** When working with large data sets, memory management is always a potential pain point. By using a streaming execution engine and buffer manager, DuckDB supports many operations on larger than memory data sets. DuckDB also aims to support queries where *intermediate* results do not fit into memory by using disk-spilling techniques.
+
+In this release support for disk-spilling techniques is further extended through the support for out-of-core hash aggregates. Now hash tables constructed during `GROUP BY` queries or `DISTINCT` operations that do not fit in memory due to the large amount of groups will spill data to disk instead of throwing an out-of-memory exception. Due to clever use of radix partitioning, performance degradation is gradual and performance cliffs are avoided. Only the subset of the table that does not fit into memory will be spilled to disk.
+
+|  memory limit | runtime   |
+|:-------------:|:---------:|
+| 6.0GB         | 6.84s     |
+| 5.0GB         | 7.39s     |
+| 4.0GB         | 9.04s     |
+| 3.0GB         | 8.76s     |
+| 2.0GB         | 8.08s     |
+| 1.0GB         | 10.27s    |
```

Comment:
These are the numbers for the initial out-of-core hash aggregate PR. It should be changed to reflect the new changes that improved our performance on ClickBench, too. Here's my edit of this section:

In this release, support for disk-spilling techniques is further extended through the support for out-of-core hash aggregates. Now, hash tables constructed during `GROUP BY` queries or `DISTINCT` operations that do not fit in memory due to a large number of unique groups will spill data to disk instead of throwing an out-of-memory exception. Due to the clever use of radix partitioning, performance degradation is gradual, and performance cliffs are avoided. Only the subset of the table that does not fit into memory will be spilled to disk.

As a result of these changes, the performance of our hash aggregate has improved in general, especially when there are many groups. For example, we compute the number of unique rows in a data set with 30 million rows and 15 columns by using the following query:
```sql
SELECT COUNT(*) FROM (SELECT DISTINCT * FROM tbl);
```
This query takes 8.52s in DuckDB 0.8.1 on a 2020 MacBook Pro. In DuckDB 0.9.0, you can expect this query to run in 2.91s!

If we keep all the data in memory, the query should use around 10GB. However, we can still complete the query if less memory is available. In the table below, we can see how the runtime is affected by lowering the memory limit:

|  memory limit |  runtime  |
|:-------------:|:---------:|
| 10.0GB        | 2.91s     |
| 9.0GB         | 3.45s     |
| 8.0GB         | 3.45s     |
| 7.0GB         | 3.47s     |
| 6.0GB         | 3.41s     |
| 5.0GB         | 3.67s     |
| 4.0GB         | 3.87s     |
| 3.0GB         | 4.20s     |
| 2.0GB         | 4.39s     |
| 1.0GB         | 4.91s     |

Reply:
Thanks, could you also write up a section for Compressed Materialization? I missed that that was this release as well.

Reply:
Sure thing!